flow/connectors/mongo/qrep.go

```diff
@@ -0,0 +1,185 @@
+package connmongo
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"go.mongodb.org/mongo-driver/v2/bson"
+	"go.mongodb.org/mongo-driver/v2/mongo"
+	"go.mongodb.org/mongo-driver/v2/mongo/options"
+
+	"github.com/PeerDB-io/peerdb/flow/connectors/utils"
+	"github.com/PeerDB-io/peerdb/flow/generated/protos"
+	"github.com/PeerDB-io/peerdb/flow/model"
+	"github.com/PeerDB-io/peerdb/flow/shared/types"
+)
+
+const MONGO_FULL_TABLE_PARTITION_ID = "mongo-full-table-partition-id"
+
+func (c *MongoConnector) GetQRepPartitions(
+	ctx context.Context,
+	config *protos.QRepConfig,
+	last *protos.QRepPartition,
+) ([]*protos.QRepPartition, error) {
+	// if no watermark column is specified, return a single partition
+	if config.WatermarkColumn == "" {
+		return []*protos.QRepPartition{
+			{
+				PartitionId:        MONGO_FULL_TABLE_PARTITION_ID,
+				Range:              nil,
+				FullTablePartition: true,
+			},
+		}, nil
+	}
+	if last != nil && last.Range != nil {
+		return nil, fmt.Errorf("not implemented")
+	}
+	parseWatermarkTable, err := utils.ParseSchemaTable(config.WatermarkTable)
+	if err != nil {
+		return nil, fmt.Errorf("unable to parse watermark table: %w", err)
+	}
+	collection := c.client.Database(parseWatermarkTable.Schema).Collection(parseWatermarkTable.Table)
+
+	// TODO: support partitioning with bucket > 1
+	// TODO: support partitioning by columns other than _id
+	numBuckets := 1
```

Comment:
should be `numDocuments / numRowsPerPartition`

Reply:
ah yeah, i want to scope this PR to just  a single snapshot/bucket (hence the `// TODO: support partitioning with bucket > 1` comment). 

Will follow-up with supporting `numDocuments/numRowsPerPartition` in a separate PR